syntax = "proto3";

import "Error.proto";

message LoginRequest{
    string username = 1;
    string password = 2;
}

message LoginResponse{
    string username = 1;
    string showname = 2;
    string token = 3;
    Error error = 4;
}

message RegisterRequest{
    string username = 1;
    string password = 2;
    string showname = 3;
}

message ChangeNameRequest{
    string username = 1;
    string name = 2;
}

message RegularResponse{
    Error error = 1;
}


